### PR TITLE
Added possibility to add default timeout in QueueMessageTemplate / QueueMessageChannel

### PR DESCRIPTION
**Pull request by dennis-menge**
Refers to gh-646 "Enable long polling in QueueMessagingTemplate #646"

### CODE REVIEW

File: spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/core/QueueMessageChannel.java [hunk 1]
```diff
@@ -219,17 +230,27 @@ private MessageAttributeValue getNumberMessageAttribute(Object messageHeaderValu
 
 	@Override
 	public Message<String> receive() {
-		return this.receive(0);
+		return this.receive(defaultTimeout);
```
Comment (eddumelendez) [inline spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/core/QueueMessageChannel.java:233]:
```suggestion
		return this.receive(this.defaultTimeout);
```

File: spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/core/QueueMessageChannel.java [hunk 2]
```diff
@@ -60,11 +60,20 @@
...
 	private final String queueUrl;
 
+	private long defaultTimeout = Long.MIN_VALUE;
+
 	public QueueMessageChannel(AmazonSQSAsync amazonSqs, String queueUrl) {
 		this.amazonSqs = amazonSqs;
 		this.queueUrl = queueUrl;
 	}
 
+	public QueueMessageChannel(AmazonSQSAsync amazonSqs, String queueUrl,
```
Comment (eddumelendez) [inline spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/core/QueueMessageChannel.java:70]: I think we can provide an setter instead another constructor

File: spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/core/QueueMessageChannel.java [hunk 3]
```diff
@@ -219,17 +230,27 @@ private MessageAttributeValue getNumberMessageAttribute(Object messageHeaderValu
...
 	@Override
 	public Message<String> receive(long timeout) {
-		ReceiveMessageResult receiveMessageResult = this.amazonSqs.receiveMessage(
-				new ReceiveMessageRequest(this.queueUrl).withMaxNumberOfMessages(1)
-						.withWaitTimeSeconds(Long.valueOf(timeout).intValue())
-						.withAttributeNames(ATTRIBUTE_NAMES)
-						.withMessageAttributeNames(MESSAGE_ATTRIBUTE_NAMES));
-		if (receiveMessageResult.getMessages().isEmpty()) {
+		ReceiveMessageResult receiveMessageResult;
+		if (timeout == Long.MIN_VALUE) { /* use Queue default timeout */
```
Comment (eddumelendez) [inline spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/core/QueueMessageChannel.java:239]: My suggestion is to do something like this

```java
ReceiveMessageRequest messageRequest = new ReceiveMessageRequest(this.queueUrl)
    .withMaxNumberOfMessages(1)
    .withAttributeNames(ATTRIBUTE_NAMES)
    .withMessageAttributeNames(MESSAGE_ATTRIBUTE_NAMES);
if (timeout != Long.MIN_VALUE) {
    messageRequest.withWaitTimeSeconds(Long.valueOf(timeout).intValue());
}
ReceiveMessageResult receiveMessageResult = this.amazonSqs.receiveMessage(messageRequest);
```

**Review by eddumelendez — CHANGES_REQUESTED** [review]
@dennis-menge thanks for this contribution! I have added a few suggestions and also wondering if we will need a configuration property `cloud.aws.sqs.default-timeout`

**Comment by dennis-menge** [comment]
Do you want to introduce a config property or would you prefer to re-use the existing `cloud.aws.sqs.listener.wait-timeout` property?

**Pull request closed without merging**